### PR TITLE
Wii Fit Trainer adjustments

### PR DESCRIPTION
**Pull request by SkewedAskew**
Hitbox comparisons: https://imgur.com/a/42kE0GQ

### Jab 1:
 - [$] Cleaned up hitboxes
 - Hand/Arm
   - [+] Damage: 2.0% -> 3.0%
   - [/] Hitlag Multiplier: 1.5x -> 1.0x
   - [-] Additional Hitstun: 4F -> 0F
 - Foot
   - [+] Damage: 2.0% -> 5.0%
   - [/] BKB: 40 -> 20
   - [/] Hitlag Multiplier: 1.5x -> 1.2x

### Jab 3:
 - Leg
   - [/] Hitlag Multiplier: 2.0x -> 1.0x
 - Foot
   - [/] Hitlag Multiplier: 2.0x -> 1.5x

### Forward Tilt:
 - [$] Attached hitboxes to body
 - [/] Hip hitbox matched to leg
 - Arm
   - [-] Hitbox Size: 4.0u -> 3.5u
 - Leg
   - [+] Hitbox Duration: F6-11 -> F5-11
   - [-] BKB: 50 -> 75
   - [+] Hitbox Size: 3.0u -> 3.5u

### Up Tilt:
 - [+] Added head and back foot hitboxes
 - [/] Priority shifted (hand > hip > arm > head > foot)
 - Arm/Hand/Head
   - [-] BKB: 40 -> 80
   - [-] Hitbox Size (hand/arm/head): 5.5/4.5u -> 4.3/4.0/2.5u
 - Hip/Back Foot
   - [-] BKB: 40 -> 80

### Dash Attack:
 - [/] Reshaped hitboxes
 - Early
   - [+] Damage: 10.0% -> 11.0%
   - [-] Shieldstun Multiplier: 1.3x -> 1.1x
 - Late
   - [+] Damage: 6.0% -> 8.0%
   - [-] Shieldstun Multiplier: 1.8x -> 1.4x

### Forward Smash:
 - [+] Added arm hitboxes
 - [-] Hand hitboxes pulled inward
 - [+] Damage (hands): 15.5% -> 16.0%
 - [-] Hitbox Size (hands/arms/knees): 4.3/3.1/3.5u -> 3.5/3.25/3.5u

### Up Smash:
 - [/] Centered hand hitboxes
 - [+] Added waist hitbox
 - Hitbox Size (waist, early/late): 5.0/3.5u

### Down Smash:
 - [-] Leg hitboxes pulled inward
 - [+] Hitbox Duration: F18-21 -> F15-18

### Neutral Air:
 - [/] Hitboxes start 1 frame later in the animation
 - [-] Hitbox Duration (hit 1/hit 2): F5-8/F9-12 -> F6-8/F9-12
 - Hit 1
   - [$] Hitboxes repositioned evenly along body
   - [-] Hitbox Size: 5.0u -> 4.0u
 - Hit 2
   - [/] Centered hitbox
   - [-] Hitbox Size: 9.0u -> 8.0u

### Forward Air:
 - Early
   - [-] Hitbox Size (hand): 5.5u -> 4.5u

### Back Air:
 - Early
   - [-] Angle: 40 -> 361
   - [/] BKB: 15/40 -> 40
   - [-] KBG: 107/102 -> 102
   - [/] Hitlag Multiplier: 1.0/1.2x -> 1.0x
 - Late
   - [+] Damage: 7.0% -> 8.0%
   - [+] Hitbox Size (foot): 3.5u -> 4.0u

### Up Air:
 - [/] Repositioned hitboxes
 - [-] Hitbox Size (hand): 5.0u -> 4.5u

### Down Air:
 - [R] Hitboxes made spherical and attached to legs
 - [-] Hitbox Size: 5.0u -> 4.0u

### Header:
 - Early
   - [-] Damage: 15.0% -> 12.0%
   - [=] BKB (ground/air): 30 -> 26/27
   - [=] KBG (ground/air): 71/49 -> 87/60

### Deep Breathing:
 - [-] Damage Dealt Multiplier: 1.2/1.25x -> 1.2x
 - [-] Damage Taken Multiplier: 0.9x -> 1.0x
 - [+] Walk Speed Multiplier: 1.17/1.2x -> 1.2x
 - [/] Ground Friction Multiplier: 1.1/1.13x -> 1.1x
 - [-] Initial Dash Speed Multiplier: 1.1/1.13x -> 1.1x
 - [-] Run Speed Multiplier: 1.1/1.13x -> 1.1x
 - [+] Airspeed Multiplier: 1.17/1.2x -> 1.2x
 - [/] Fallspeed/Gravity Multiplier: 1.17/1.2x -> 1.2x

### CODE REVIEW

File: fighters/common/src/function_hooks/get_param.rs [hunk 1]
```diff
@@ -69,6 +70,20 @@ pub unsafe fn get_param_int_hook(x0: u64, x1: u64, x2 :u64) -> i32 {
             }
         }
 
+        else if fighter_kind == *FIGHTER_KIND_WIIFIT {
+            if VarModule::is_flag(boma_reference.object(), vars::wiifit::instance::DEEP_BREATHING_COOLDOWN)
+            && x1 == hash40("param_special_lw") {
+                if x2 == 0x21a6281763 {
+                    println!("smaller h");
```
Comment (CSharpM7) [inline fighters/common/src/function_hooks/get_param.rs]: leftover println

File: fighters/common/src/function_hooks/get_param.rs [hunk 2]
```diff
@@ -69,6 +70,20 @@ pub unsafe fn get_param_int_hook(x0: u64, x1: u64, x2 :u64) -> i32 {
...
+                    return 300;
+                }
+                else if x2 == 0x21011c971a {
+                    println!("bigger h");
```
Comment (CSharpM7) [inline fighters/common/src/function_hooks/get_param.rs]: leftover println

File: fighters/wiifit/src/acmd/ground.rs [hunk 3]
```diff
@@ -7,24 +6,14 @@ unsafe fn wiifit_attack_11_game(fighter: &mut L2CAgentBase) {
...
+        ATTACK(fighter, 1, 0, Hash40::new("arml"), 3.0, 361, 25, 0, 20, 2.75, -3.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_G_d, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_PUNCH, *ATTACK_REGION_PUNCH);
+        ATTACK(fighter, 2, 0, Hash40::new("footr"), 5.0, 361, 130, 0, 20, 3.0, -0.5, 0.0, 0.0, None, None, None, 1.2, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_B, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA_d, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_KICK, *ATTACK_REGION_KICK);
```
Comment (CSharpM7) [inline fighters/wiifit/src/acmd/ground.rs:12]: Should this still have 130 KBG with the increased damage? Or is reducing the BKB from 40 to 20 enough?

File: fighters/wiifit/src/acmd/ground.rs [hunk 4]
```diff
@@ -101,28 +90,44 @@ unsafe fn wiifit_attack_dash_game(fighter: &mut L2CAgentBase) {
...
-        ATTACK(fighter, 1, 0, Hash40::new("bust"), 6.0, 65, 80, 0, 75, 3.0, 3.5, 0.0, 3.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_PUNCH, *ATTACK_REGION_PUNCH);
-        ATK_SET_SHIELD_SETOFF_MUL_arg4(fighter, 0, 1, 2, 1.8);
+        ATTACK(fighter, 0, 0, Hash40::new("top"), 6.0, 65, 80, 0, 75, 4.5, 0.0, 7.0, 8.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_M, *COLLISION_SOUND_ATTR_PUNCH, *ATTACK_REGION_PUNCH);
```
Comment (CSharpM7) [inline fighters/wiifit/src/acmd/ground.rs]: PR patch notes say the damage is 8.0%, but this shows the damage as unchanged

**Review by CSharpM7 — CHANGES_REQUESTED** [review]

File: fighters/wiifit/src/acmd/ground.rs (hunk 3 above)
Comment (SkewedAskew) [reply fighters/wiifit/src/acmd/ground.rs:12]: Yeah, the BKB decrease was enough

File: fighters/wiifit/src/acmd/ground.rs (hunk 4 above)
Comment (SkewedAskew) [reply fighters/wiifit/src/acmd/ground.rs]: Oops

**Review by CSharpM7 — APPROVED** [review]

**Pull request merged**